java-client/src/main/java/org/opensearch/client/opensearch/_types/aggregations/CompositeDateHistogramAggregationSource.java

```diff
@@ -0,0 +1,192 @@
+import org.opensearch.client.json.JsonpMapper;
+import org.opensearch.client.json.ObjectBuilderDeserializer;
+import org.opensearch.client.json.ObjectDeserializer;
+import org.opensearch.client.opensearch._types.Time;
+import org.opensearch.client.util.ObjectBuilder;
+
+public class CompositeDateHistogramAggregationSource extends CompositeValuesSource {
+
+    @Nullable
+    private final Time calendarInterval;
+
+    @Nullable
+    private final Time fixedInterval;
+
+    private final long offset;
+
+    private final String zoneId;
+
+    private CompositeDateHistogramAggregationSource(Builder builder) {
+        super(builder);
+        this.calendarInterval = builder.calendarInterval;
+        this.fixedInterval = builder.fixedInterval;
+        this.offset = builder.offset;
+        this.zoneId = builder.zoneId;
+    }
+
+    public static CompositeDateHistogramAggregationSource of(Function<Builder, ObjectBuilder<CompositeDateHistogramAggregationSource>> fn) {
+        return fn.apply(new Builder()).build();
+    }
+
+    /**
+     * API name: {@code calendar_interval}
+     */
+    @Nullable
+    public final Time calendarInterval() {
+        return this.calendarInterval;
+    }
+
+    /**
+     * API name: {@code fixed_interval}
+     */
+    @Nullable
+    public final Time fixedInterval() {
+        return this.fixedInterval;
+    }
+
+    /**
+     * Required - API name: {@code offset}
+     */
+    public final long offset() {
+        return this.offset;
+
+    }
+
+    /**
+     * Required - API name: {@code zone_id}
+     */
+    public final String zoneId() {
+        return this.zoneId;
+    }
+
+    /**
+     * Serialize this object to JSON.
+     */
+    @Override
+    public void serialize(JsonGenerator generator, JsonpMapper mapper) {
+        generator.writeStartObject();
+        serializeInternal(generator, mapper);
+        generator.writeEnd();
+    }
+
+    protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
+        super.serializeInternal(generator, mapper);
+        if (this.calendarInterval != null) {
+            generator.writeKey("calendar_interval");
+            this.calendarInterval.serialize(generator, mapper);
+
+        }
+        if (this.fixedInterval != null) {
+            generator.writeKey("fixed_interval");
+            this.fixedInterval.serialize(generator, mapper);
+
+        }
+        generator.writeKey("offset");
+        generator.write(this.offset);
+
+        generator.writeKey("zone_id");
+        generator.write(this.zoneId);
+    }
+
+    /**
+     * Builder for {@link CompositeDateHistogramAggregationSource}.
+     */
+
+    public static class Builder extends CompositeValuesSource.AbstractBuilder<Builder>
+        implements
+            ObjectBuilder<CompositeDateHistogramAggregationSource> {
+
+        @Nullable
+        private Time calendarInterval;
+
+        @Nullable
+        private Time fixedInterval;
+
+        private long offset;
```

Comment:
@Nullable?

Reply:
Yeah this can be null I think, don't see any restrictions on the server end.